Задача №1  
Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16.
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

In [45]:
import scipy.stats as st
import numpy as np

my = 16
z = st.norm.ppf(0.975)
x = 80
n = 256

In [46]:
[x-z*(my/(n**0.5)), x+z*(my/(n**0.5))]

[78.04003601545995, 81.95996398454005]

Задача №2  
В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные:  
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1  
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

Находим t - критерий:

In [67]:
i = 10
t = st.t.ppf(0.95, i-1)
t

1.8331129326536335

Находим среднее значение выборки:

In [68]:
k = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
sr = np.mean(k)
sr

6.590000000000001

Рассчитываем сигму по выборке:

In [73]:
mu = np.std(k, ddof=1)
mu

0.4508017549014448

In [74]:
[sr - t * (mu/(i**0.5)), sr + t * (mu/(i**0.5))]

[6.328678694353238, 6.8513213056467634]

Ответ:  Истинное значение величины X с доверительной вероятностью 95% находится в интервале [6.328678694353238, 6.8513213056467634]

Задача №3  
Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170  
Рост матерей  178, 165, 165, 173, 168, 155, 160, 164, 178, 175  
Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.

In [56]:
t1 = st.t.ppf(0.95, 18)
t1

1.7340636066175354

Находим среднее значение 1й выборки:

In [57]:
x1 = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])
sr1 = np.mean(x1)
sr1

166.2

Находим среднее значение 2й выборки:

In [75]:
x2 = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])
sr2 = np.mean(x2)
sr2

168.1

Находим дельту между 1й и 2й выборками:

In [59]:
delta = sr1-sr2
delta

-1.9000000000000057

Рассчитываем дисперсию для 1й выборки:

In [60]:
disp1 = np.std(x1, ddof=1)**2
disp1

91.06666666666665

Рассчитываем дисперсию для 2й выборки:

In [61]:
disp2 = np.std(x2, ddof=1)**2
disp2

60.10000000000001

Рассчитываем среднюю дисперсию для обоих выборок:

In [62]:
disp = (disp1 + disp2)**0.5
disp

12.294985427671993

Рассчитываем ошибку:

In [63]:
se = (disp/10 + disp/10)**0.5
se

1.5681189640886302

Применяем формулу для интервальной оценки разности средних арифм.:

In [64]:
[delta - t1 * se, delta + t1 * se]

[-4.61921802647289, 0.8192180264728779]